python/activator/raw.py

```diff
@@ -149,36 +157,142 @@ def is_path_consistent(oid: str, visit: FannedOutVisit) -> bool:
     return False
 
 
+def check_for_snap(
+    client,
+    bucket: str,
+    microservice: str,
+    instrument: str,
+    group: int,
+    snap: int,
+    detector: int,
+) -> str | None:
+    """Search for new raw files matching a particular data ID.
+
+    The search is performed in the active image bucket or in a raw
+    image microservice, if one is available.
+
+    Parameters
+    ----------
+    client : `S3.Client`
+        The client object with which to do the search.
+    bucket : `str`
+        The name of the bucket in which to search.
+    microservice : `str`
+        The URI of an optional microservice to assist the search.
+    instrument, group, snap, detector
+        The data ID to search for.
+
+    Returns
+    -------
+    name : `str` or `None`
+        The raw's object key within ``bucket``, or `None` if no file
+        was found. If multiple files match, this function logs an error
+        but returns one of the files anyway.
+    """
+    prefix = get_prefix_from_snap(microservice, instrument, group, detector, snap)
+    if not prefix:
+        return None
+    _log.debug(f"Checking for '{prefix}'")
+    response = client.list_objects_v2(Bucket=bucket, Prefix=prefix)
```

Comment:
The microservice guarantees that the file is present, so this call is not strictly required.